feat: separate two forms for handeling update for basic account details

Waishnav opened the pull request:
Solves #153 

This PR separates the My Account settings page into two distinct forms:
1. **Basic Details Form**: For updating fields like first name and last name.
2. **Password Update Form**: For changing the user's password.

![image](https://github.com/user-attachments/assets/cfeaa89b-592f-4953-88da-586d33179503)

**Note for future**: we can update the first form for other fields in User models.

loststoner commented:
Hey @Waishnav , Split the UI into two tabs: one for Personal Settings and the other for Security

Example:
![Screenshot from 2025-01-09 11-59-35](https://github.com/user-attachments/assets/cec0c790-d5cc-4fbd-934f-fb2f520163e1)

We might need to add more fields in the future, so it's better to keep personal details and security settings in separate sections.

Waishnav commented:
> Hey @Waishnav , Split the UI into two tabs: one for Personal Settings and the other for Security

![image](https://github.com/user-attachments/assets/aad1e2ee-0f5c-40b7-94ad-432ea233acd9)
Added it, let me know if there is any changes needed?

riddick-oe commented on src/users/views/my_account.py:
```diff
@@ -0,0 +1,117 @@
+        })
+        
+        return render(request, self.template_name, {
+            'profile_form': profile_form,
```
Let's use double quotes `"` for strings wherever we can, to be consistent.

Waishnav replied:
I've fixed those in new commit, thanks' for letting me know about the string convention!

riddick-oe commented on src/users/views/my_account.py:
```diff
@@ -0,0 +1,117 @@
+            try:
+                # Only update first_name and last_name
+                user = request.user
+                user.first_name = form.cleaned_data['first_name']
```
https://github.com/factryflow/factryflow/pull/265/files#r1908306051

Same goes for the rest of the file for `"` strings.

riddick-oe commented on src/users/forms/user.py:
```diff
@@ -143,3 +143,42 @@ def clean(self):
+
+    class Meta:
+        model = User
+        fields = ['first_name', 'last_name']
```
Let's use double quotes " for strings wherever we can, to be consistent.
Check the rest of the file, as well.
